src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -1181,7 +1187,7 @@ private fun processLexicalDeclarations(
 
     fun processPattern(pattern: RsPat, processor: RsResolveProcessor): Boolean {
         val boundNames = PsiTreeUtil.findChildrenOfType(pattern, RsPatBinding::class.java)
-            .filter { it.reference.resolve() == null }
+            .filter { it.parent is RsPatField || !it.isReferenceToConstant }
```

Comment:
Could you comment on this line?

Reply:
AFAIR, it fixes resolve in match branches. Same syntax can be both variable binding and reference to constant. Before this PR it could only resolve to constant, so simple null check was enough.